File: Java/programmers/스택큐/다리를지나는트럭.java
package java. programmers.스택큐;

import java.util.LinkedList;
import java.util.Queue;

public class 다리를지나는트럭 {
	 public int solution(int bridge_length, int weight, int[] truck_weights) {
	        Queue<Integer> queue = new LinkedList<>();
	        int sum = truck_weights[0];
	        queue.offer(truck_weights[0]);
	        int answer = 0;
	        int time =0;
	        int i = 1;
	        while(!queue.isEmpty()){
	           if(i>=truck_weights.length) break;
	           if(queue.size()== bridge_length){
	               sum-= queue.poll();
	           }
	           if(i<truck_weights.length && sum+truck_weights[i]<=weight){
	              sum+=truck_weights[i];
	              queue.offer(truck_weights[i]);
	              i++;
	           } else{
	               queue.offer(0);
	           }
	            answer++;
	        }
	        answer+= ++bridge_length;
	        return answer;
	    }
}

File: JavaScript/programmers/다리를지나는트럭.js
function solution(bridge_length, weight, truck_weights) {
    var answer = 0;
    const queue = [];
    let sum = truck_weights[0];
    let cnt = 1;
    queue.push(truck_weights[0]);
    while(queue.length){
        if(cnt >= truck_weights.length) break;
        if(queue.length === bridge_length){
            sum -= queue.shift();
        }
        if(cnt < truck_weights.length && weight >= sum+ truck_weights[cnt]){
            sum += truck_weights[cnt];
            queue.push(truck_weights[cnt]);
            cnt++;
        }else{
            queue.push(0);
        }
        answer++;
    }
    answer += bridge_length + 1;
    return answer;
}